src/js/options.js

```diff
@@ -224,11 +230,37 @@ function removeAllData() {
   }
 }
 
+//TODO Do we want confirmations before or success/failure messages after for these?
+function downloadWhitelist() {
+  chrome.runtime.sendMessage({type: "downloadWhitelist"}, () => {
+    //Set a timeout because this needs to happen just a tiny bit later than it was without.
+    setTimeout(function() {
```

Comment:
We need to refresh the whitelist so that it correctly displays the updated values, and in my testing, the refresh seemed to be happening before the data actually got set (the whitelist wouldn't be refreshed after downloading, but manually calling this function from the console on the settings page would update it).

Reply:
I now believe this was actually caused by the way that the onMessage event handler executes, which was cutting off the callback from the `chrome.storage.sync.get` function.  (The solution to that is to return true from the event handler; this was found when debugging the success/failure alert always displaying failure.)